presto-main/src/main/java/com/facebook/presto/sql/analyzer/AggregationAnalyzer.java

```diff
@@ -533,4 +565,46 @@ public Boolean process(Node node, @Nullable Void context)
             return super.process(node, context);
         }
     }
+
+    /**
+     * Verifies that no query node output columns are referenced from ORDER BY expressions.
+     * Only FROM columns are allowed within ORDER BY expressions.
+     */
+    private class VerifyOrderByAggregationVisitor
+            extends DefaultTraversalVisitor<Void, Void>
+    {
+        @Override
+        protected Void visitIdentifier(Identifier node, Void context)
+        {
+            verifyField(node, QualifiedName.of(node.getName()));
```

Comment:
it probably also needs verification for columnReference (not lambda) if https://github.com/prestodb/presto/pull/7210 lands before